### PR TITLE
Adds update complete support to the MemoryConfigClient

### PR DESCRIPTION
**Pull request by balazsracz**
Adds support for sending UPDATE_COMPLETE, REBOOT, FREEZE, and UNFREEZE commands from the MemoryConfigClient to a remote node.
Adds unittests including checking the case when the remote node actually reboots as part of these sequences.

Misc bugfixes:
- fixes a flakiness in the Bootlaoder client test
- makes sure that the datagram reboot command recognizes the initialization complete message even if dst only has the alias filled in and not the node ID
- performance improvement for message send flow to avoid a dynamic allocation and copy of the payload string.

### CODE REVIEW

File: src/openlcb/MemoryConfigClient.hxx [hunk 1]
```diff
@@ -135,9 +220,19 @@ public:
...
+            Defs::OPENMRN_TIMEOUT, //< Timeout waiting for ack/nack.
+    };
+
 private:
     Action entry() override
     {
+        request()->resultCode = OPERATION_PENDING;
         switch (request()->cmd)
         {
         case MemoryConfigClientRequest::CMD_READ:
```
Comment (bakerstu) [inline src/openlcb/MemoryConfigClient.hxx]: Not a big deal, but the style guide says that these case statements should be indented.

File: src/openlcb/MemoryConfigClient.hxx [hunk 2]
```diff
@@ -135,9 +220,19 @@ public:
     {
     }
 
+    enum ResultCodes
```
Comment (bakerstu) [inline src/openlcb/MemoryConfigClient.hxx]: Add doxygen comment.

File: src/openlcb/MemoryConfigClient.hxx [hunk 3]
```diff
@@ -135,9 +220,19 @@ public:
...
+    {
+        // Internal error codes generated by the send flow
+        OPERATION_PENDING =
+            DatagramClient::OPERATION_PENDING, //< cleared when done is called.
```
Comment (bakerstu) [inline src/openlcb/MemoryConfigClient.hxx]: These doxygen comments need another '/' to be valid.

File: src/openlcb/MemoryConfigClient.hxx (hunk 1 above)
Comment (balazsracz) [reply src/openlcb/MemoryConfigClient.hxx]: done

File: src/openlcb/MemoryConfigClient.hxx (hunk 2 above)
Comment (balazsracz) [reply src/openlcb/MemoryConfigClient.hxx]: done

File: src/openlcb/MemoryConfigClient.hxx (hunk 3 above)
Comment (balazsracz) [reply src/openlcb/MemoryConfigClient.hxx]: done

**Pull request merged**